src/main/java/plugily/projects/villagedefense/user/data/MysqlManager.java

```diff
@@ -86,20 +104,8 @@ public void saveStatistic(User user, StatsStorage.StatisticType stat) {
 
   @Override
   public void saveAllStatistic(User user) {
-    StringBuilder update = new StringBuilder(" SET ");
-    for(StatsStorage.StatisticType stat : StatsStorage.StatisticType.values()) {
-      if(!stat.isPersistent()) {
-        continue;
-      }
-      if(update.toString().equalsIgnoreCase(" SET ")) {
-        update.append(stat.getName()).append('=').append(user.getStat(stat));
-      }
-      update.append(", ").append(stat.getName()).append('=').append(user.getStat(stat));
-    }
-    String finalUpdate = update.toString();
-
-    Bukkit.getScheduler().runTaskAsynchronously(plugin, () ->
-        database.executeUpdate("UPDATE " + getTableName() + finalUpdate + " WHERE UUID='" + user.getUniqueId().toString() + "';"));
+    String updateQuery = getUpdateQuery(user);
+    Bukkit.getScheduler().runTaskAsynchronously(plugin, () -> database.executeUpdate(updateQuery));
```

Comment:
I think `getUpdateQuery` can be performed safely in asynchronously.
```suggestion
    Bukkit.getScheduler().runTaskAsynchronously(plugin, () -> database.executeUpdate(getUpdateQuery(user)));
```